### PR TITLE
Add SDF integrator tests, further cleanup

### PR DESCRIPTION
**Pull request by helenol**
Bug fixes:
 - Fix the double slices (sometimes slices were 2 pts thick) (FINALLY!)
 - Improve ESDF generation, fix some small bugs with it
 - Fix a corner case of TSDF integration

Tests:
 - Tests for both ESDF and TSDF generation using simulation world

ESDF Integrator:
 - Use neighbor tools instead :)

### CODE REVIEW

**Comment by helenol** [comment]
test this please (jenkins misconfigured, marks builds failed on "max error" in cout output.)

File: voxblox/include/voxblox/core/common.h [hunk 1]
```diff
@@ -78,6 +78,7 @@ typedef AlignedVector<Triangle> TriangleVector;
 typedef kindr::minimal::QuatTransformationTemplate<FloatingPoint>
     Transformation;
 typedef kindr::minimal::RotationQuaternionTemplate<FloatingPoint> Rotation;
+typedef Eigen::Quaternion<FloatingPoint> Quaternion;
```
Comment (ZacharyTaylor) [inline voxblox/include/voxblox/core/common.h]: Would it be better to use kindr::minimal::RotationQuaternionTemplate<FloatingPoint>::Implementation
that way all rotations stay as minkindr types

File: voxblox/include/voxblox/integrator/esdf_integrator.h [hunk 2]
```diff
@@ -35,7 +36,7 @@ class EsdfIntegrator {
     FloatingPoint default_distance_m = 2.0;
     // For cheaper but less accurate map updates: the minimum difference in
     // a voxel distance, before the change is propagated.
-    FloatingPoint min_diff_m = 0.001;
+    FloatingPoint min_diff_m = 0.0; //0.001;
```
Comment (ZacharyTaylor) [inline voxblox/include/voxblox/integrator/esdf_integrator.h]: Was this causing issues? from memory even a tiny value here can speed things up a lot

File: voxblox/include/voxblox/utils/layer_utils.h [hunk 3]
```diff
@@ -13,7 +13,8 @@ namespace voxblox {
 namespace utils {
 
 template <typename VoxelType>
-bool isSameVoxel(const VoxelType& voxel_A, const VoxelType& voxel_B) {
+bool isSameVoxel(const VoxelType& /* voxel_A */,
```
Comment (ZacharyTaylor) [inline voxblox/include/voxblox/utils/layer_utils.h]: why remove the variable names?

File: voxblox/include/voxblox/utils/neighbor_tools.h [hunk 4]
```diff
@@ -30,7 +30,7 @@ class NeighborTools {
     CHECK_NE(voxels_per_side_, 0u);
   }
 
-  void getNeighborsAndDistances(
+  void getNeighborIndicesAndDistances(
```
Comment (ZacharyTaylor) [inline voxblox/include/voxblox/utils/neighbor_tools.h]: While I prefer "Indicies" we have been going with "Indexes" in voxblox

File: voxblox/src/integrator/esdf_integrator.cc [hunk 5]
```diff
@@ -18,6 +18,12 @@ EsdfIntegrator::EsdfIntegrator(const Config& config,
   CHECK_NEAR(esdf_layer_->voxel_size(), tsdf_layer_->voxel_size(), 1e-6);
 
   open_.setNumBuckets(config_.num_buckets, config_.max_distance_m);
+
+  neighbor_tools_.setLayer(esdf_layer);
+
+  std::cout << "ESDF integrator settings: min_dist: " << config_.min_distance_m
```
Comment (ZacharyTaylor) [inline voxblox/src/integrator/esdf_integrator.cc]: Maybe LOG(INFO) instead of std::cout?

File: voxblox/src/integrator/esdf_integrator.cc [hunk 6]
```diff
@@ -94,11 +100,21 @@ void EsdfIntegrator::addNewRobotPosition(const Point& position) {
...
         esdf_voxel.hallucinated = true;
-        pushNeighborsToOpen(kv.first, voxel_index);
         updated_blocks_.insert(kv.first);
       }
     }
   }
+
+  // Now push all the neighbors to open, now that all the relevant neighbors
+  // are allocated.
+  // Don't need to check the free set, as the occupied voxel list contains also
```
Comment (ZacharyTaylor) [inline voxblox/src/integrator/esdf_integrator.cc]: grammar

File: voxblox/src/integrator/esdf_integrator.cc [hunk 7]
```diff
@@ -704,29 +737,36 @@ void EsdfIntegrator::processOpenSet() {
...
-                   distance_to_neighbor) {
-          // Do nothing, we good.
-        } else if (neighbor_voxel.distance < 0.0 &&
-                   esdf_voxel.distance > distance_to_neighbor) {
-          // OK now the other case is if it's 2 totally different signs...
-          neighbor_voxel.distance = -distance_to_neighbor;
+          // ESDF voxel not in fixed band, and is outside an obstacle, while the
+          // neighbor voxel is inside an obstacle.
+        } else if (neighbor_voxel.distance < 0.0) {
+          // neighbor_voxel.distance = -distance_to_neighbor;
```
Comment (ZacharyTaylor) [inline voxblox/src/integrator/esdf_integrator.cc]: remove commented line?

File: voxblox/include/voxblox/utils/layer_utils.h (hunk 3 above)
Comment (helenol) [reply voxblox/include/voxblox/utils/layer_utils.h]: This keeps giving warnings in clang, since we have warnings for unused variables on.

File: voxblox/include/voxblox/integrator/esdf_integrator.h (hunk 2 above)
Comment (helenol) [reply voxblox/include/voxblox/integrator/esdf_integrator.h]: Oh good call, turned it off while debugging.

Comment (helenol) [reply voxblox/include/voxblox/integrator/esdf_integrator.h]: Quickly double-checked, error completely unaffected but computations 1.2x as fast or so. Definitely putting it back in.
